src/htr2hpc/api_client.py

```diff
@@ -0,0 +1,329 @@
+            details = ""
+            if info.status == "error":
+                details = info.error
+            logger.error(f"Bad request {details}")
+        elif resp.status_code == requests.codes.not_found:
+            logger.error("Error: not found")
+        elif resp.status_code == requests.codes.unauthorized:
+            logger.error("Error: not authorized")
+        else:
+            # TODO: error handling / logging for other cases?
+            # (useful for dev if nothing else...)
+            logger.error(resp.status_code)
+            logger.error(resp.content)
+
+    def current_user(self):
+        """Get information about the current user account"""
+        api_url = "users/current/"
+        return to_namedtuple("user", self._make_request(api_url).json())
+
+    def models(self):
+        """paginated list of models"""
+        api_url = "models/"
+        resp = self._make_request(api_url)
+        return ResultsList(result_type="model", **resp.json())
+
+    def update_model(self, model_id: int, model_file: pathlib.Path):
+        """Update an existing model record with a new model file."""
+        api_url = f"models/{model_id}/"
+        with open(model_file, "rb") as mfile:
+            files = {"file": mfile}
+            data = {
+                "name": "updated model",
+                "job": "Segment",  # required; get from existing record?
+                # file_size (int)  - set from pathlib object using .stat().st_size
+                # versions ?
+                # accuracy_percent
+            }
+            resp = self._make_request(api_url, method="PUT", files=files, data=data)
+        # on successful update, returns the model object
+        return to_namedtuple("model", resp.json())
+
+    def documents(self):
+        """paginated list of documents"""
+        api_url = "documents/"
+        resp = self._make_request(api_url)
+        return ResultsList(result_type="document", **resp.json())
+
+    def document(self, document_id: int):
+        """details for a single document"""
+        api_url = f"documents/{document_id}/"
+        resp = self._make_request(api_url)
+        return to_namedtuple("document", resp.json())
+
+    def document_export(
+        self, document_id: int, transcription_id: int, include_images: bool = False
+    ):
+        """request a document export be compiled for download"""
+        api_url = f"documents/{document_id}/export/"
+        # export form requires a region_types list, which
+        # is a based on block type ids for this document
+        # and may also include "Undefined" and "Orphan"
+        document_info = self.document(document_id)
+        block_types = [block.pk for block in document_info.valid_block_types]
+        block_types.extend(["Undefined", "Orphan"])
+
+        data = {
+            "transcription": transcription_id,  # could be multiple(?)
+            "file_format": "alto",
+            "include_images": True,
```

Comment:
Should this use the input arg `include_images`?
```suggestion
            "include_images": include_images,
```